File: javaapp/src/main/java/com/scandit/shelf/javaapp/MainActivity.java
package com.scandit.shelf.javaapp;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.scandit.shelf.sdk.authentication.Authentication;
import com.scandit.shelf.javaapp.ui.base.NavigationFragment;
import com.scandit.shelf.javaapp.ui.login.LoginFragment;
import com.scandit.shelf.javaapp.ui.storeselection.StoreSelectionFragment;

/**
 * The main Activity that hosts all Fragments in the sample app.
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);
        if (savedInstanceState == null) {
            NavigationFragment initFragment = Authentication.isAuthenticated()
                    ? StoreSelectionFragment.newInstance()
                    : LoginFragment.newInstance();
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.fragment_container, initFragment)
                    .commitNow();
        }
    }
}

File: kotlinapp/src/main/kotlin/com/scandit/shelf/kotlinapp/MainActivity.kt
package com.scandit.shelf.kotlinapp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.scandit.shelf.kotlinapp.ui.login.LoginFragment
import com.scandit.shelf.kotlinapp.ui.storeselection.StoreSelectionFragment
import com.scandit.shelf.sdk.authentication.Authentication

/**
 * The main Activity that hosts all Fragments in the sample app.
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
        if (savedInstanceState == null) {
            val initFragment = if (Authentication.isAuthenticated) {
                StoreSelectionFragment.newInstance()
            } else {
                LoginFragment.newInstance()
            }
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, initFragment)
                .commitNow()
        }
    }
}
